packages/flutter/test/material/carousel_test.dart

```diff
@@ -1977,6 +1977,244 @@ void main() {
+      expect(leadingIndex, equals(3));
+
+      controller.animateToItem(
+        1,
+        duration: const Duration(milliseconds: 200),
+        curve: Curves.linear,
+      );
+
+      await tester.pumpAndSettle();
+      expect(controller.leadingItem, equals(1));
+      expect(leadingIndex, equals(1));
+    });
+
+    testWidgets(
+      'CarouselView.weighted shows correct item after animation with symmetric flexWeights',
+      (WidgetTester tester) async {
+        final controller = CarouselController();
+        addTearDown(controller.dispose);
+        var leadingIndex = 0;
+
+        await tester.pumpWidget(
+          MaterialApp(
+            home: Scaffold(
+              body: CarouselView.weighted(
+                flexWeights: const [2, 5, 2],
+                controller: controller,
+                onIndexChanged: (int index) => leadingIndex = index,
+                children: List.generate(6, (i) => Text('Item $i')),
+              ),
+            ),
+          ),
+        );
+        await tester.pumpAndSettle();
+
+        controller.animateToItem(
+          4,
+          duration: const Duration(milliseconds: 200),
+          curve: Curves.linear,
+        );
+        await tester.pumpAndSettle();
+
+        expect(controller.leadingItem, equals(4));
+        expect(leadingIndex, equals(4));
+        expect(find.text('Item 4'), findsOneWidget);
+
+        final Rect viewport = viewportRect(tester);
+        final Rect rect4 = tester.getRect(getItem(4));
+
+        expect(rect4.top, greaterThanOrEqualTo(viewport.top));
+        expect(rect4.bottom, lessThanOrEqualTo(viewport.bottom));
+      },
+    );
+
+    testWidgets(
+      'CarouselView.weighted shows correct item after animation with asymmetric flexWeights',
+      (WidgetTester tester) async {
+        final controller = CarouselController();
+        addTearDown(controller.dispose);
+        var leadingIndex = 0;
+
+        await tester.pumpWidget(
+          MaterialApp(
+            home: Scaffold(
+              body: CarouselView.weighted(
+                flexWeights: const [1, 2, 3, 4],
+                controller: controller,
+                onIndexChanged: (int index) => leadingIndex = index,
+                children: List.generate(6, (i) => Text('Item $i')),
+              ),
+            ),
+          ),
+        );
+        await tester.pumpAndSettle();
+
+        controller.animateToItem(
+          2,
+          duration: const Duration(milliseconds: 200),
+          curve: Curves.linear,
+        );
+        await tester.pumpAndSettle();
+
+        expect(controller.leadingItem, equals(2));
+        expect(leadingIndex, equals(2));
+        expect(find.text('Item 2'), findsOneWidget);
+
+        final Rect viewport = viewportRect(tester);
+        final Rect rect2 = tester.getRect(getItem(2));
+
+        expect(rect2.top, greaterThanOrEqualTo(viewport.top));
+        expect(rect2.bottom, lessThanOrEqualTo(viewport.bottom));
```

Comment:
Same here and everywhere else. greaterThanOrEqualTo and lessThanOrEqualTo are not accurate enough to show the expected results.